eth_defi/event_reader/ankr.py

```diff
@@ -0,0 +1,89 @@
+import requests
+import json
+from enum import Enum
+
+from eth_defi.event_reader.block_header import BlockHeader
+
+
+class AnkrSupportedBlockchain(Enum):
+    eth = "eth"
+    bsc = "bsc"
+    polygon = "polygon"
+    fantom = "fantom"
+    arbitrum = "arbitrum"
+    avalanche = "avalanche"
+    syscoin = "syscoin"
+
+
+def make_block_request_ankr(endpoint_url: str = "https://rpc.ankr.com/multichain/79258ce7f7ee046decc3b5292a24eb4bf7c910d7e39b691384c7ce0cfb839a01/", start_block: int | str = "latest", end_block: int | str = "latest", blockchain: AnkrSupportedBlockchain | None = None, request_id: int = 1) -> list[dict]:
+    """Fetch blocks from Ankr API
+
+    :param endpoint_url: URL of Ankr API endpoint. Should be multichain endpoint.
+
+    :param start_block: Block number to start fetching from. Can be an int or "latest".
+
+    :param end_block: Block number to end fetching at. Can be an int or "latest".
+
+    :param blockchain: Blockchain to fetch blocks from. Must be of type AnkrSupportedBlockchain.
+
+    :return: List of blocks in JSON format.
+    """
+    if start_block == "latest":
+        end_block = "latest"
+
+    assert type(start_block) == int or start_block == "latest", "start_block must be an int or 'latest'"
+    assert type(end_block) == int or end_block == "latest", "end_block must be an int or None"
+    assert isinstance(blockchain, AnkrSupportedBlockchain), "blockchain must be of type AnkrSupportedBlockchain"
+
+    headers = {
+        "accept": "application/json",
+        "content-type": "application/json",
+    }
+
+    data = {
+        "jsonrpc": "2.0",
+        "method": "ankr_getBlocks",
+        "params": {
+            "blockchain": blockchain.value,
+            "fromBlock": start_block,
+            "toBlock": end_block,
+            "includeTxs": False,
+            "includeLogs": False,
+        },
+        "id": request_id,
+    }
+
+    result = requests.post(endpoint_url, headers=headers, json=data)
+
+    j = result.json()
+
+    blocks = j["result"]["blocks"]
+
+    return blocks
+
+
+def extract_timestamps_ankr_get_block(
```

Comment:
This must use Web3 instance, not direct `endpoint_url`, because otherwise the function won't be compatible or in line with the rest of the library.

For making custom RPC requests, see `anvil.py` for examples.

If you need to fish out the `endpoint_url` from web3 object you need to do:

```python
provider = web3.provider
if isinstance(provider, HTTPProvider):
    endpoint_uri = provider.endpoint_uri
```

As the provider may also use `WebsocketProvider` and in that case we should fail with an exception until a special support is added.